services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +43,40 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
-
-  return ERR_CODE_SUCCESS;
+  if(event == NULL){
+    return ERR_CODE_INVALID_QUEUE_MSG;
+  } 
+  if(xQueueSend(thermalMgrQueueHandle,(void*) event, (TickType_t) 0) == pdPASS){
+    return ERR_CODE_SUCCESS;
+  }
+  return ERR_CODE_UNKNOWN;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t interrupt;
+  interrupt.type = THERMAL_MGR_EVENT_OVER_TEMP;
```

Comment:
it would be better to name the type something else `THERMAL_MGR_EVENT_OVER_TEMP` is misleading since the interrupt would happen when we return to safe operating conditions as well